examples/wasm-example/src/routes/signing.rs

```diff
@@ -0,0 +1,392 @@
+    },
+    ExtrinsicFailed(anyhow::Error),
+}
+
+impl Component for SigningExamplesComponent {
+    type Message = Message;
+
+    type Properties = ();
+
+    fn create(ctx: &Context<Self>) -> Self {
+        ctx.link().send_future(OnlineClient::<PolkadotConfig>::new().map(|res| {
+            match res {
+                Ok(online_client) => Message::OnlineClientCreated(online_client),
+                Err(err) => Message::Error(anyhow!("Online Client could not be created. Make sure you have a local node running:\n{err}")),
+            }
+        }));
+        SigningExamplesComponent {
+            message: "".to_string(),
+            stage: SigningStage::CreatingOnlineClient,
+            online_client: None,
+            remark_call_bytes: vec![],
+        }
+    }
+
+    fn update(&mut self, ctx: &Context<Self>, msg: Self::Message) -> bool {
+        match msg {
+            Message::OnlineClientCreated(online_client) => {
+                self.online_client = Some(online_client);
+                self.stage = SigningStage::EnterMessage;
+                self.set_message("Hello".into());
+            }
+            Message::ChangeMessage(message) => {
+                self.set_message(message);
+            }
+            Message::RequestAccounts => {
+                self.stage = SigningStage::RequestingAccounts;
+                ctx.link().send_future(get_accounts().map(
+                    |accounts_or_err| match accounts_or_err {
+                        Ok(accounts) => Message::ReceivedAccounts(accounts),
+                        Err(err) => Message::Error(err),
+                    },
+                ));
+            }
+            Message::ReceivedAccounts(accounts) => {
+                self.stage = SigningStage::SelectAccount(accounts);
+            }
+            Message::Error(err) => self.stage = SigningStage::Error(err.to_string()),
+            Message::SignWithAccount(i) => {
+                if let SigningStage::SelectAccount(accounts) = &self.stage {
+                    let account = accounts.get(i).unwrap();
+                    let account_address = account.address.clone();
+                    let account_source = account.source.clone();
+                    let account_id: AccountId32 = account_address.parse().unwrap();
+                    web_sys::console::log_1(&account_id.to_string().into());
+
+                    self.stage = SigningStage::Signing(account.clone());
+
+                    let remark_call = polkadot::tx()
+                        .system()
+                        .remark(self.message.as_bytes().to_vec());
+
+                    let api = self.online_client.as_ref().unwrap().clone();
+
+                    ctx.link()
+                        .send_future(
+                            async move {
+                                let partial_extrinsic =
+                                    match api.tx().create_partial_signed(&remark_call, &account_id, Default::default()).await {
```

Comment:
I'm a little wary of exposing `extrinsic_params` from `PartialExtrinsic` (and generally of exposing the `BaseExtrinsicParams` props). In part because I don't want us to be "committed" to those details in a public API (I like the idea that we can provide an interface to build extrinsic params etc, but don't think people should have any access to the internals so that we are free to change it, as we will be doing soon). 

I also don't think that it's necessary to expose them, because they have all been provided beforehand by the user or are available on the client anyway. So I'd do something more explicit, and gather each of the properties that you need, like:

```rust
let spec_version = api.runtime_version().spec_version;
let transaction_version = api.runtime_version().transaction_version;
let genesis_hash = api.genesis_hash();
let mortality_checkpoint = api.genesis_hash();
let era = subxt::config::Era::Immortal;
let method = api.tx().call_data(&remark_call)?;
let nonce = api.tx().account_nonce(&account_id).await?;
// ...
```

(Maybe eg `extension_signature_for_partial_extrinsic` takes in the client and call data and fetches the rest, or something)